### PR TITLE
feat: add `array/base/last`

### LINKED ISSUE #858: [RFC]: Add `@stdlib/array/base/last`
### Description

This RFC proposes adding support for returning the last element of an array-like object, with support for the accessor protocol.

```javascript
last( [ 1, 2, 3 ] );
// returns 3

z = new Complex128Array( [ 1, 2, 3, 4 ] );
last( z );
// returns <Complex128>
```

Package: `@stdlib/array/base/last`
alias: `last`

### Related Issues

None.

### Questions

No.

### Other

See [`@stdlib/array/base/copy`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/base/copy) for an example of a similar package which supports the accessor protocol.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Infinage**
## Description

> What is the purpose of this pull request?

This pull request:

-   Adds support for returning the last element of an array-like object

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #858

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/array/base/last/README.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# last
+
+> Returns the last element from an array-like object, with support for the accessor protocol.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/README.md]:
```suggestion
> Return the last element of an array-like object.
```

File: lib/node_modules/@stdlib/array/base/last/README.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+## Usage
+
+```javascript
+var last = require( '@stdlib/array/base/last' );
+```
+
+#### last( x )
+
+Returns the element at the last index from an array-like object.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/README.md]:
```suggestion
Returns the last element of an array-like object.
```

File: lib/node_modules/@stdlib/array/base/last/README.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+```javascript
+var x = [ 1, 2, 3 ];
+
+var out = last( x );
+// returns 3
+
+var bool = ( out === 1 );
+// returns false
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/array/base/last/README.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+<!-- eslint no-undef: "error" -->
+
+```javascript
+var Complex64Array = require( '@stdlib/array/complex64' );
+var realf = require( '@stdlib/complex/realf' );
+var imagf = require( '@stdlib/complex/imagf' );
+var last = require( '@stdlib/array/base/last' );
+
+// Create a complex number array:
+var arr = new Complex64Array( 10 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/README.md]:
```suggestion
var arr = new Complex64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 ] );
```

File: lib/node_modules/@stdlib/array/base/last/README.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+// Returns the last element from the complex number array: <Complex64>
+var out = last( arr );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/README.md]:
```suggestion
// Return the last element:
var out = last( arr );
// returns <Complex64>
```

File: lib/node_modules/@stdlib/array/base/last/README.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+var re = realf( out );
+// returns 0.0
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/README.md]:
```suggestion
// returns 5.0
```

File: lib/node_modules/@stdlib/array/base/last/README.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+var im = imagf( out );
+// returns 0.0
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/README.md]:
```suggestion
// returns 6.0
```

File: lib/node_modules/@stdlib/array/base/last/README.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+
+console.log( '%d + %di', re, im );
+// => '0 + 0i'
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/README.md]:
```suggestion
// => '5 + 6i'
```

File: lib/node_modules/@stdlib/array/base/last/benchmark/benchmark.js [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+// MODULES //
+
+var bench = require( '@stdlib/bench' );
+var discreteUniform = require( '@stdlib/random/base/discrete-uniform' ).factory;
+var Complex64Array = require( '@stdlib/array/complex64' );
+var pow = require( '@stdlib/math/base/special/pow' );
+var filledBy = require( '@stdlib/array/filled-by' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var pkg = require( '@stdlib/array/base/last/package.json' ).name;
+var last = require('@stdlib/array/base/last/lib');
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/benchmark/benchmark.js]:
```suggestion
var last = require( './../lib' );
```

File: lib/node_modules/@stdlib/array/base/last/benchmark/benchmark.js [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+
+// MODULES //
+
+var bench = require( '@stdlib/bench' );
+var discreteUniform = require( '@stdlib/random/base/discrete-uniform' ).factory;
+var Complex64Array = require( '@stdlib/array/complex64' );
+var pow = require( '@stdlib/math/base/special/pow' );
+var filledBy = require( '@stdlib/array/filled-by' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var pkg = require( '@stdlib/array/base/last/package.json' ).name;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/benchmark/benchmark.js]:
```suggestion
var pkg = require( './../package.json' ).name;
```

File: lib/node_modules/@stdlib/array/base/last/benchmark/benchmark.js [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+'use strict';
+
+// MODULES //
+
+var bench = require( '@stdlib/bench' );
+var discreteUniform = require( '@stdlib/random/base/discrete-uniform' ).factory;
+var Complex64Array = require( '@stdlib/array/complex64' );
+var pow = require( '@stdlib/math/base/special/pow' );
+var filledBy = require( '@stdlib/array/filled-by' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/benchmark/benchmark.js]:
```suggestion
var isnanf = require( '@stdlib/math/base/assert/is-nanf' );
```

File: lib/node_modules/@stdlib/array/base/last/benchmark/benchmark.js [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+
+/**
+* Creates a benchmark function.
+*
+* @private
+* @param {PositiveInteger} len - array length
+* @returns {Function} benchmark function
+*/
+function createBenchmark( len ) {
+	var buf = filledBy( len, 'float64', rand );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/benchmark/benchmark.js]:
```suggestion
	var buf = filledBy( len*2, 'float32', rand );
```

File: lib/node_modules/@stdlib/array/base/last/benchmark/benchmark.js [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+	* @param {Benchmark} b - benchmark instance
+	*/
+	function benchmark( b ) {
+		var out;
+		var i;
+
+		b.tic();
+		for ( i = 0; i < b.iterations; i++ ) {
+			out = last( arr );
+			if ( isnan( out ) ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/benchmark/benchmark.js]:
```suggestion
			if ( isnanf( out ) ) {
```

File: lib/node_modules/@stdlib/array/base/last/benchmark/benchmark.js [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+				b.fail( 'should not return NaN' );
+			}
+		}
+		b.toc();
+		if ( isnan( out ) ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/benchmark/benchmark.js]:
```suggestion
		if ( isnanf( out ) ) {
```

File: lib/node_modules/@stdlib/array/base/last/benchmark/benchmark.js [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+	var i;
+
+	min = 1; // 10^min
+	max = 6; // 10^max
+
+	for ( i = min; i <= max; i++ ) {
+		len = pow( 10, i );
+
+		f = createBenchmark( len );
+		bench( pkg, f );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/benchmark/benchmark.js]:
```suggestion
		bench( pkg+':len='+len, f );
```

File: lib/node_modules/@stdlib/array/base/last/docs/repl.txt [hunk 16]
```diff
@@ -0,0 +1,22 @@
+
+{{alias}}( arr )
+    Returns the element at the last index from an array-like object.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/docs/repl.txt]:
```suggestion
    Returns the last element of an array-like object.
```

File: lib/node_modules/@stdlib/array/base/last/docs/repl.txt [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+
+    Parameters
+    ----------
+    arr: ArrayLikeObject
+        Input array.
+
+    Returns
+    -------
+    out: Element
+        Element at the last index.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/docs/repl.txt]:
```suggestion
    out: any
        Last element.
```

File: lib/node_modules/@stdlib/array/base/last/docs/types/index.d.ts [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+*/
+
+// TypeScript Version: 2.0
+
+/// <reference types="@stdlib/types"/>
+
+import { Collection } from '@stdlib/types/object';
+
+/**
+* Returns the last element from an array-like object.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/docs/types/index.d.ts]:
```suggestion
* Returns the last element of an array-like object.
```

File: lib/node_modules/@stdlib/array/base/last/docs/types/index.d.ts [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+*
+* @param arr - input array
+* @returns element value
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/docs/types/index.d.ts]:
```suggestion
* @returns last element
```

File: lib/node_modules/@stdlib/array/base/last/examples/index.js [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+var out = last( arr );
+
+var re = realf( out );
+// returns 0.0
+
+var im = imagf( out );
+// returns 0.0
+
+console.log( '%d + %di', re, im );
+// => '0 + 0i'
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/examples/index.js]:
```suggestion
var arr = new Complex64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 ] );

// Return the last element:
var out = last( arr );
// returns <Complex64>

var re = realf( out );
// returns 5.0

var im = imagf( out );
// returns 6.0

console.log( '%d + %di', re, im );
// => '5 + 6i'
```

File: lib/node_modules/@stdlib/array/base/last/lib/index.js [hunk 21]
```diff
@@ -0,0 +1,40 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+/**
+* Return the last element from an array-like object.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/lib/index.js]:
```suggestion
* Return the last element of an array-like object.
```

File: lib/node_modules/@stdlib/array/base/last/lib/index.js [hunk 22]
```diff
@@ -0,0 +1,40 @@
...
+*
+* @module @stdlib/array/base/last
+*
+* @example
+* var last = require( '@stdlib/array/base/last' );
+*
+* last( [ 1, 2, 3 ] );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/lib/index.js]:
```suggestion
* var out = last( [ 1, 2, 3 ] );
```

File: lib/node_modules/@stdlib/array/base/last/lib/main.js [hunk 23]
```diff
@@ -0,0 +1,68 @@
...
+var isAccessorArray = require( '@stdlib/array/base/assert/is-accessor-array' );
+var accessorGetter = require( '@stdlib/array/base/accessor-getter' );
+var getter = require( '@stdlib/array/base/getter' );
+var dtype = require( '@stdlib/array/dtype' );
+
+
+// MAIN //
+
+/**
+* Returns the last element from an array-like object.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/lib/main.js]:
```suggestion
* Returns the last element of an array-like object.
```

File: lib/node_modules/@stdlib/array/base/last/lib/main.js [hunk 24]
```diff
@@ -0,0 +1,68 @@
...
+*
+* @param {Collection} arr - input array
+* @returns {Object} - element value
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/lib/main.js]:
```suggestion
* @returns {*} - last element
```

File: lib/node_modules/@stdlib/array/base/last/lib/main.js [hunk 25]
```diff
@@ -0,0 +1,68 @@
...
+	dt = dtype( arr );
+
+	// Resolve an accessor for retrieving input array elements:
+	if ( isAccessorArray( arr ) ) {
+		get = accessorGetter( dt );
+	} else {
+		get = getter( dt );
+	}
+
+	// Get the number of elements to return the element at the last index:
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/lib/main.js:57]:
```suggestion
	// Get the number of elements in order to resolve the last index:
```

File: lib/node_modules/@stdlib/array/base/last/lib/main.js [hunk 26]
```diff
@@ -0,0 +1,68 @@
...
+* @param {Collection} arr - input array
+* @returns {Object} - element value
+*
+* @example
+* var out = last( [ 1, 2, 3 ] );
+* // returns 3
+*/
+function last( arr ) {
+	var out;
+	var len;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/lib/main.js]:
```suggestion
```

File: lib/node_modules/@stdlib/array/base/last/lib/main.js [hunk 27]
```diff
@@ -0,0 +1,68 @@
...
+	} else {
+		get = getter( dt );
+	}
+
+	// Get the number of elements to return the element at the last index:
+	len = arr.length;
+
+	// Return the element at the last index (0 - Based indices)
+	out = get( arr, len - 1 );
+	return out;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/lib/main.js]:
```suggestion
	// Return the last element:
	return get( arr, arr.length - 1 );
```

File: lib/node_modules/@stdlib/array/base/last/lib/main.js [hunk 28]
```diff
@@ -0,0 +1,68 @@
...
+*
+* @param {Collection} arr - input array
+* @returns {Object} - element value
+*
+* @example
+* var out = last( [ 1, 2, 3 ] );
+* // returns 3
+*/
+function last( arr ) {
+	var out;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/lib/main.js]:
```suggestion
```

File: lib/node_modules/@stdlib/array/base/last/package.json [hunk 29]
```diff
@@ -0,0 +1,61 @@
+{
+  "name": "@stdlib/array/base/last",
+  "version": "0.0.0",
+  "description": "Returns the last element from an array-like object.",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/package.json]:
```suggestion
  "description": "Return the last element of an array-like object.",
```

File: lib/node_modules/@stdlib/array/base/last/test/test.js [hunk 30]
```diff
@@ -0,0 +1,83 @@
...
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof last, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function returns the last element from an array-like object', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/test/test.js]:
```suggestion
tape( 'the function returns the last element of an array-like object', function test( t ) {
```

File: lib/node_modules/@stdlib/array/base/last/test/test.js [hunk 31]
```diff
@@ -0,0 +1,83 @@
...
+
+	x = [ 1, 2, 3 ];
+	expected = 3;
+	actual = last( x );
+
+	t.equal( actual, expected, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the function returns the last element from an array-like object (accessors)', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/test/test.js]:
```suggestion
tape( 'the function returns the last element of an array-like object (accessors)', function test( t ) {
```

File: lib/node_modules/@stdlib/array/base/last/test/test.js [hunk 32]
```diff
@@ -0,0 +1,83 @@
...
+		'imagf': 4.0
+	};
+	var actual;
+	var x;
+
+	x = new Complex64Array( [ 1.0, 2.0, 3.0, 4.0 ] );
+	actual = last( x );
+
+	t.strictEqual( realf(actual), expected.realf, 'returns expected value' );
+	t.strictEqual( imagf(actual), expected.imagf, 'returns expected value' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/test/test.js]:
```suggestion
	var actual;
	var x;

	x = new Complex64Array( [ 1.0, 2.0, 3.0, 4.0 ] );
	actual = last( x );

	t.strictEqual( realf(actual), 3.0, 'returns expected value' );
	t.strictEqual( imagf(actual), 4.0, 'returns expected value' );
```

File: lib/node_modules/@stdlib/array/base/last/test/test.js [hunk 33]
```diff
@@ -0,0 +1,83 @@
...
+		'imagf': 2.0
+	};
+	var actual;
+	var x;
+
+	x = new Complex64Array( [ 1.0, 2.0 ] );
+	actual = last( x );
+
+	t.strictEqual( realf(actual), expected.realf, 'returns expected value' );
+	t.strictEqual( imagf(actual), expected.imagf, 'returns expected value' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/test/test.js]:
```suggestion
tape( 'the function returns `undefined` if provided an empty array', function test( t ) {
	var actual = last( [] );
	t.strictEqual( actual, void 0, 'returns expected value' );
```

File: lib/node_modules/@stdlib/array/base/last/test/test.js [hunk 34]
```diff
@@ -0,0 +1,83 @@
...
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/test/test.js:69]:
```suggestion
```

File: lib/node_modules/@stdlib/array/base/last/test/test.js [hunk 35]
```diff
@@ -0,0 +1,83 @@
...
+	t.end();
+});
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/test/test.js:74]:
```suggestion
});

tape( 'the function returns `undefined` if provided an empty array (accessors)', function test( t ) {
	var actual = last( new Complex64Array( 0 ) );
	t.strictEqual( actual, void 0, 'returns expected value' );
	t.end();
});
```

File: lib/node_modules/@stdlib/array/base/last/lib/main.js [hunk 36]
```diff
@@ -0,0 +1,61 @@
...
+* @param {Collection} arr - input array
+* @returns {*} - last element
+*
+* @example
+* var out = last( [ 1, 2, 3 ] );
+* // returns 3
+*/
+function last( arr ) {
+	var get;
+	var dt;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/lib/main.js:44]:
```suggestion
	var idx;
	var dt;
```

File: lib/node_modules/@stdlib/array/base/last/lib/main.js [hunk 37]
```diff
@@ -0,0 +1,61 @@
...
+	dt = dtype( arr );
+
+	// Resolve an accessor for retrieving input array elements:
+	if ( isAccessorArray( arr ) ) {
+		get = accessorGetter( dt );
+	} else {
+		get = getter( dt );
+	}
+	// Return the last element:
+	return get( arr, arr.length - 1 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/lib/main.js]:
```suggestion
	// Resolve the last index:
	idx = arr.length - 1;
	
	// Return the last element:
	if ( idx < 0 ) {
		return;
	}
	return get( arr, arr.length - 1 );
```

File: lib/node_modules/@stdlib/array/base/last/lib/main.js [hunk 38]
```diff
@@ -0,0 +1,68 @@
...
+
+	// Resolve an accessor for retrieving input array elements:
+	if ( isAccessorArray( arr ) ) {
+		get = accessorGetter( dt );
+	} else {
+		get = getter( dt );
+	}
+	// Resolve the last index:
+	idx = arr.length - 1;
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/lib/main.js]:
```suggestion

```

File: lib/node_modules/@stdlib/array/base/last/lib/main.js [hunk 39]
```diff
@@ -0,0 +1,67 @@
...
+	// Resolve an accessor for retrieving input array elements:
+	if ( isAccessorArray( arr ) ) {
+		get = accessorGetter( dt );
+	} else {
+		get = getter( dt );
+	}
+	// Resolve the last index:
+	idx = arr.length - 1;
+	// Return the last element:
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/lib/main.js:58]:
```suggestion

	// Return the last element:
```

File: lib/node_modules/@stdlib/array/base/last/lib/main.js [hunk 40]
```diff
@@ -0,0 +1,68 @@
...
+		get = getter( dt );
+	}
+	// Resolve the last index:
+	idx = arr.length - 1;
+
+	// Return the last element:
+	if ( idx < 0 ) {
+		return;
+	}
+	return get( arr, arr.length - 1 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/last/lib/main.js]:
```suggestion
	return get( arr, idx );
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Infinage!

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Infinage!

**Comment by Infinage** [comment]
Thanks for the review and the merge @kgryte! I will take better care of the edge cases and add better examples the next time. Most of the changes you did made sense. But I am just curious about these two changes:

1. Why are we multiplying the len by 2 in the `createBenchmark()`?

image

2. I am assuming this one is done for better documentation of the test results. I noticed that in the other packages we had `benchmark.length.js` in addition to the `benchmark.js` file. I wasn't able to understand the difference between those two files. However do we need to rename this file to `benchmark.length.js` since I see `pkg+':len='+len` only in files with this name?

image

**Comment by kgryte** [comment]
@Infinage Regarding changes,

1. You need to multiply by `2` in order for the `Complex64Array` to have the expected length. A complex number array has interleaved real and imaginary components. Meaning, for every complex number element, the element is comprised of a real number and a complex number which are adjacent in memory. Hence, when allocating the underlying buffer, we need to double the storage capacity.

2. Yes, in general, for benchmarks which measuring performance as a function of array length, we use the `.length.js` suffix to make clear that the benchmark considers input argument size when measuring perf. We use `pkg+':len='+len` as the file is dynamically creating benchmark functions, and we need to be able to distinguish the benchmark runs from one another in the TAP output.

For now, you do **not** need to rename the file. I can take care of that on my end.

Hope this answers your questions.

**Comment by Infinage** [comment]
Yes it does, thanks!

**Pull request merged**